front/lib/api/files/tool_output.ts

```diff
@@ -0,0 +1,51 @@
+import { Readable } from "stream";
+import { pipeline } from "stream/promises";
+
+import type { Authenticator } from "@app/lib/auth";
+import { FileResource } from "@app/lib/resources/file_resource";
+export async function internalCreateToolOutputCsvFile({
+  auth,
+  title,
+  content,
+  contentType,
+}: {
+  auth: Authenticator;
+  title: string;
+  content: string;
+  contentType: "text/csv";
+}): Promise<FileResource> {
+  const workspace = auth.getNonNullableWorkspace();
+  const user = auth.getNonNullableUser();
+
+  const fileResource = await FileResource.makeNew({
+    workspaceId: workspace.id,
+    userId: user.id,
+    contentType,
+    fileName: title,
+    fileSize: Buffer.byteLength(content),
+    useCase: "toolOutput",
+  });
+
+  // Write both the "original" and "processed" versions simultaneously
+
+  await Promise.all([
```

Comment:
Can we directly call `maybeApplyPreProcessing`, so we keep the preprocessing/file creation in one single place? This function also handles the `markAsReady()`/`markAsFailed`.

Reply:
The problem with the pre processing code is that I have to first upload the original, then read it again then re-upload the processed, even though they will always be identical. Here I feel it might make more sense to do everything in one pass. I could move this function to the processing file maybe ?

Reply:
Yep in this case, if we can consolidate in one file 🙏.

Reply:
We might still want to handle failures to upload and update the file accordingly.